Framework/Beamline/test/SpectrumInfoTest.h

```diff
@@ -0,0 +1,95 @@
+#ifndef MANTID_BEAMLINE_SPECTRUMINFOTEST_H_
+#define MANTID_BEAMLINE_SPECTRUMINFOTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidBeamline/SpectrumInfo.h"
+#include "MantidBeamline/SpectrumDefinition.h"
+
+using namespace Mantid::Beamline;
+
+class SpectrumInfoTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static SpectrumInfoTest *createSuite() { return new SpectrumInfoTest(); }
+  static void destroySuite(SpectrumInfoTest *suite) { delete suite; }
+
+  void test_copy() {
+    const SpectrumInfo source(7);
+    const auto copy(source);
+    TS_ASSERT_EQUALS(copy.size(), 7);
+  }
+
+  void test_move() {
+    SpectrumInfo source(7);
+    const auto moved(std::move(source));
+    TS_ASSERT_EQUALS(moved.size(), 7);
+    TS_ASSERT_EQUALS(source.size(), 0);
+  }
+
+  void test_assign() {
+    const SpectrumInfo source(7);
+    SpectrumInfo assignee(1);
+    assignee = source;
+    TS_ASSERT_EQUALS(assignee.size(), 7);
+  }
+
+  void test_move_assign() {
+    SpectrumInfo source(7);
+    SpectrumInfo assignee(1);
+    assignee = std::move(source);
+    TS_ASSERT_EQUALS(assignee.size(), 7);
+    TS_ASSERT_EQUALS(source.size(), 0);
+  }
+
+  void test_size() {
+    TS_ASSERT_EQUALS(SpectrumInfo(0).size(), 0);
+    TS_ASSERT_EQUALS(SpectrumInfo(1).size(), 1);
+  }
+
+  void test_spectrumDefinition() {
+    SpectrumInfo info(1);
+    TS_ASSERT_EQUALS(info.spectrumDefinition(0).size(), 0);
+  }
+
+  void test_setSpectrumDefinition() {
+    SpectrumDefinition def;
+    def.add(7, 5);
+    TS_ASSERT_EQUALS(def.size(), 1);
+    SpectrumInfo info(3);
+    info.setSpectrumDefinition(1, def);
+    TS_ASSERT_EQUALS(info.spectrumDefinition(1)[0],
+                     (std::pair<size_t, size_t>(7, 5)));
+    TS_ASSERT_EQUALS(def.size(), 1);
+  }
```

Comment:
Should there be an explicit test for the copy on write behaviour - i.e. copy info, add a new `SpectrumDefinition`, verify they are different?